huaweicloud/resource_huaweicloud_compute_servergroup.go

```diff
@@ -129,51 +137,87 @@ func resourceComputeServerGroupUpdate(d *schema.ResourceData, meta interface{})
 
 		for _, v := range membersToAdd.List() {
 			var addMemberOpts servergroups.MemberOpts
-			addMemberOpts.InstanceID = v.(string)
+			instanceId := v.(string)
+			// The ECS instances do not support other operations when binding server groups.
+			config.MutexKV.Lock(instanceId)
+			addMemberOpts.InstanceID = instanceId
 			if err := servergroups.UpdateMember(ecsClient, addMemberOpts, "add_member", d.Id()).ExtractErr(); err != nil {
+				// Release the ECS instance after the binding fails.
+				config.MutexKV.Unlock(instanceId)
 				return fmtp.Errorf("Error to add a instance to ECS server group, err=%s", err)
 			}
+			// Release the ECS instance after the binding is complete.
+			config.MutexKV.Unlock(instanceId)
 		}
 
 		for _, v := range membersToRemove.List() {
-			instanceID := v.(string)
-			server, err := cloudservers.Get(ecsClient, instanceID).Extract()
+			instanceId := v.(string)
+			// Any operations are not supported when an ECS instance is unbound from a server group.
+			config.MutexKV.Lock(instanceId)
+			server, err := cloudservers.Get(ecsClient, instanceId).Extract()
```

Comment:
I think we can query the ECS directly whether it has multiple operations or not.

Reply:
we can't guarantee that the disk will not enter other attach or detach operations at the moment after the query operation.

Reply:
done.